### DISCUSSION TITLE
4 Wire 54123 Egg Control - Just a bit of clarification required

### DISCUSSION THREAD

**Discussion by tyler56895** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#320] [Q&A]
Hi Everybody!
What a great little project this is.
I have my own Cancun egg (6 wire #54286), running Firmware 2022-04-11, working perfectly. I set the heater to come on at 00:35 and off at 04:25, heating it to 40°C every night, at a maximum cost of 40p, on Octopus Go. With my CleverSpa thermal cover, it only drops 1 or 2°C by 20:30, which i think is pretty good.

Like a idiot, I offered to WiFi enable my sister #54123 egg!
After a bit of trail and error, I managed to get it working but its different to mine, i'd just like some clarification that its correct. 
I've selected NO54123 in model.h and to make any remote changes, I have to select the "take Control" slider first, which blocks any input form the egg and to operate any timed commands, the setting must remain in "Take Control" position, otherwise the events don't' occur? The black box that displays the LED temperature on my version is constantly blank on the NO54123 version.

Also when I was experimenting, with the 2 data wires and the Model.h file settings, i struggled to know if the firmware was actually changing. In the end, in Visual Studio, I erased the flash, then saved the file withe model.h modified, closed Visual Studio, rebuilt the file, then downloaded the 4 wire application. Is it necessary to erase the ESP8266 before doing a new download?
![Lay z spa](https://user-images.githubusercontent.com/70145008/177151162-f76ad9c6-b85c-4c23-9ac6-6e1c1217176a.jpg)

**Comment by chunkysteveo** [comment]
If it's a 4 wire egg, you're right with your findings, they are not as interactive as the 6 wire ones... basically the pump unit does not communicate back to the display unit, it's all one way traffic. So the 4 wire versions have this 'hack' version that takes control of the pump unit - cutting the display unit out of the loop. You either have the display unit managing the pump OR the ESP8266 unit - via the Take Control 'God' mode.

It's the best us 4-wire people have!

**Comment by chunkysteveo** [comment]
The 'display' in the web GUI is left blank as the ESP8266 doesn't really know what the display is showing. It could show the temperature, but that would just be cosmetic as the actual seven segment display may be showing an error value, which the ESP8266 isn't capable of knowing.

**Comment by visualapproach** [comment]
You beat me to it steve, thanks! Let me add to that; you don't need to wipe flash on the esp. just upload the new stuff. You can see model and firmware date on the web ui.

**Comment by tyler56895** [comment]
Thanks for the conformation both. I noticed the Model number and firmware at the bottom of this version.

Not sure if you want to add my Cancun egg (6 wire #54286), running Firmware 2022-04-11, working perfectly, to the list of working models?

**Reply by visualapproach** [reply]
Certainly!

**Comment by tyler56895** [comment]
Not sure if I should ask this as a new question or if its ok to keep it in my original post?

Any way, when you set a timer to bring the heater on at night, would it not be possible to initially set another function to "take control" and then hand control  back after switching the heater off again? This would allow the local controls to always be available during the day, but I'm guessing this isn't possible or as easy as it sounds.

**Reply by visualapproach** [reply]
I think it's trivial. I also think I will be able to make some of the display buttons work under user control which will make your request a bit obsolete. And I'm thinking it will be more obvious to the user what device is in charge. Thanks for your input!
